prov/udp/src/udpx.h

```diff
@@ -88,6 +87,8 @@ struct udpx_ep_entry {
 	uint8_t			resv[sizeof(size_t) - 2];
 };
 
+#define UDPX_MAX_MSG_SIZE(mtu) ((mtu) - 28)
```

Comment:
What's the 28 here?

Reply:
Combined size of UDP and IP headers. It's a step forward from the previous magic value of 1472 for the max message size, but I don't know of a convenient constant value or values I could otherwise use. I could make some local macro value definitions for clarity, I suppose.

Reply:
Where do you get that size from? Are you talking about the size of msghdr?

Reply:
Well, I inferred that the value of `max_msg_size` needed to be smaller than the MTU because of the default, fixed value that `max_msg_size` had previously, namely 1472. Since 1472 is the maximum data size for a UDP packet in normally-sized Ethernet frames, I took a leap and figured that the relation between MTU size and `max_msg_size` was due to the UDP and IP headers. If there's more to it than that, I'd be happy to work out a better approach with some guidance.

Reply:
That makes sense, thanks! I couldn't find a better way to do it, you can just leave as is, but I do appreciate the local macro to make it clearer, thanks!